client/lib/config/test/index.js

```diff
@@ -50,5 +46,37 @@ describe( 'index', () => {
 
 			expect( result ).to.be.true;
 		} );
+
+		context( 'error cases', () => {
+			const NODE_ENV = process.env.NODE_ENV;
+			const fakeKey = 'where did all the errors go?';
+
+			afterEach( () => process.env.NODE_ENV = NODE_ENV );
+
+			it( `should throw an error when given key doesn't exist (NODE_ENV == development)`, () => {
+				process.env.NODE_ENV = 'development';
+
+				expect( () => config( fakeKey ) ).to.throw( ReferenceError );
+			} );
+
+			it( `should not throw an error when given key doesn't exist (NODE_ENV != development)`, () => {
+				const envs = [
+					'client',
+					'desktop',
+					'horizon',
+					'production',
+					'stage',
+					'test',
+					'wpcalypso',
+				];
+
+				envs.forEach( env => {
+					process.env.NODE_ENV = env;
```

Comment:
It would be nice to revert `process.env.NODE_ENV` to value set before this test suite is executed.